rust/processor/src/processors/default_processor.rs

```diff
@@ -380,4 +414,67 @@ impl ProcessorTrait for DefaultProcessor {
     fn connection_pool(&self) -> &PgDbPool {
         &self.connection_pool
     }
+
+    fn parse_transactions(
+        &self,
+        transactions: Vec<Transaction>,
+        start_version: u64,
+        end_version: u64,
+        _: Option<u64>,
+    ) -> anyhow::Result<Option<ProcessingParseResult>> {
```

Comment:
is the idea this would return something that implements `QueryModelBatchTrait` ?

Reply:
after processing we should return `ProcessingParseResult` which has a field for an array of `QueryModelBatchTrait`. the reason i did it this way instead of returning a vec of `QueryModelBatchTrait`, each with its own start and end versions is because it makes gap detection really difficult.